### PR TITLE
Use api/v0.4.0 after releasing it with connect-go

### PR DESCRIPTION
**Pull request by simonswine**
This is a follow-up to #2909

### CODE REVIEW

File: go.work.sum [hunk 1]
```diff
@@ -973,6 +969,8 @@ github.com/boynux/squid-exporter v1.10.5-0.20230618153315-c1fae094e18e h1:C1vYe7
 github.com/boynux/squid-exporter v1.10.5-0.20230618153315-c1fae094e18e/go.mod h1:8NpZERGK+R9DGuZqqsKfnf2qI/rh7yBT8End29IvgNA=
 github.com/bradfitz/gomemcache v0.0.0-20190913173617-a41fca850d0b h1:L/QXpzIa3pOvUGt1D1lA5KjYhPBAN/3iWdP7xeFS9F0=
 github.com/bradfitz/gomemcache v0.0.0-20190913173617-a41fca850d0b/go.mod h1:H0wQNHz2YrLsuXOZozoeDmnHXkNCRmMW0gwFWDfEZDA=
+github.com/bufbuild/connect-go v1.4.1 h1:6usL3JGjKhxQpvDlizP7u8VfjAr1JkckcAUbrdcbgNY=
+github.com/bufbuild/connect-go v1.4.1/go.mod h1:9iNvh/NOsfhNBUH5CtvXeVUskQO1xsrEviH7ZArwZ3I=
```
Comment (simonswine) [inline go.work.sum:973]: I really don't know why this is there

Comment (cyriltovena) [reply go.work.sum:973]: 🤷

Comment (simonswine) [reply go.work.sum:973]: I would say it is better than before the PR, the resulting binary is free of bufbuild:

```
go version -m ./pyroscope | grep bufbuild
```

Comment (korniltsev) [reply go.work.sum:973]: https://github.com/grafana/agent/blob/v0.35.4/go.mod#L21

Comment (simonswine) [reply go.work.sum]: It does, thanks!

Comment (korniltsev) [reply go.work.sum]: I also noticed that there are entries which I can delete manually and `go` command does not add it back :D

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**